src/core/agent/services/connectionService.ts

```diff
@@ -252,10 +261,81 @@ class ConnectionService extends AgentService {
 
   @OnlineOnly
   async deleteConnectionById(id: string): Promise<void> {
-    await this.props.signifyClient.contacts().delete(id);
+    await this.props.signifyClient
+      .contacts()
+      .delete(id)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (!/404/gi.test(status)) {
+          throw error;
+        }
+      });
     await this.connectionStorage.deleteById(id);
   }
 
+  async markConnectionPendingDelete(id: string) {
+    const connectionProps = await this.connectionStorage.findById(id);
+    if (!connectionProps) return;
+    connectionProps.pendingDeletion = true;
+    await this.connectionStorage.update(connectionProps);
+
+    this.props.eventEmitter.emit<ConnectionRemovedEvent>({
+      type: EventTypes.ConnectionRemoved,
+      payload: {
+        connectionId: id,
+      },
+    });
+  }
+
+  async getConnectionsPendingDeletion() {
+    const connections = await this.connectionStorage.findAllByQuery({
+      pendingDeletion: true,
+    });
+
+    return connections.map((connection) => connection.id);
+  }
+
+  async getConnectionsPending() {
+    const connections = await this.connectionStorage.findAllByQuery({
+      pending: true,
+      groupId: undefined,
+    });
+
+    return connections.map((connection) => connection.id);
+  }
+
+  @OnlineOnly
+  async reconnectPendingConnectionById(id: string) {
```

Comment:
I don't get we're attempting to do here. Here's what we do when resolving normal connection QR codes:
- Create a pending connection in the DB, emit and event
- Capture the event and resolve the OOBI and track long running operation (once complete, connection is no longer pending as normal)
- On start-up, check for any pending connections in case we crash

Reply:
So we need to simplify the `connectByOobiUrl` function to just create the pending connection IF it's not related to multi-sig (`groupId`)

Reply:
This might slightly clash with Martin's PR - for this to join with Martin's PR (depending on who merges first), the flow will be the same except there will be an additional step to do `contacts().update()` when the long running operation completes

Reply:
1. Create a pending connection in the DB
2. Emit this event
3. When we capture the event, trigger resolving the OOBI
4. On start-up, resolve the OOBI of any pending connection
5. When the long running operations complete, we mark the connection as complete

So resolveOobi should be called by the event handler or on start-up, and not from connectByOobiUrl.